opensearch_py_ml/ml_commons/rag_pipeline/rag/AIConnectorHelper.py

```diff
@@ -0,0 +1,512 @@
+
+    @staticmethod
+    def get_opensearch_domain_info(region, domain_name):
+        """
+        Retrieve the OpenSearch domain endpoint and ARN based on the domain name and region.
+        """
+        try:
+            opensearch_client = boto3.client("opensearch", region_name=region)
+            response = opensearch_client.describe_domain(DomainName=domain_name)
+            domain_status = response["DomainStatus"]
+            domain_endpoint = domain_status.get("Endpoint")
+            domain_arn = domain_status["ARN"]
+            return domain_endpoint, domain_arn
+        except Exception as e:
+            print(f"Error retrieving OpenSearch domain info: {e}")
+            return None, None
+
+    def get_ml_auth(self, create_connector_role_name):
+        """
+        Obtain AWS4Auth credentials for ML API calls using the specified IAM role.
+        """
+        create_connector_role_arn = self.iam_helper.get_role_arn(
+            create_connector_role_name
+        )
+        if not create_connector_role_arn:
+            raise Exception(f"IAM role '{create_connector_role_name}' not found.")
+
+        temp_credentials = self.iam_helper.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            "es",
+            session_token=temp_credentials["SessionToken"],
+        )
+        return awsauth
+
+    def create_connector(self, create_connector_role_name, payload):
+        """
+        Create a connector in OpenSearch using the specified role and payload.
+        Reusing create_standalone_connector from Connector class.
+        """
+        # Assume role and create a temporary authenticated OS client
+        create_connector_role_arn = self.iam_helper.get_role_arn(
+            create_connector_role_name
+        )
+        temp_credentials = self.iam_helper.assume_role(create_connector_role_arn)
+        temp_awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            "es",
+            session_token=temp_credentials["SessionToken"],
+        )
+
+        parsed_url = urlparse(self.opensearch_domain_url)
+        host = parsed_url.hostname
+        port = parsed_url.port or (443 if parsed_url.scheme == "https" else 9200)
+
+        temp_os_client = OpenSearch(
+            hosts=[{"host": host, "port": port}],
+            http_auth=temp_awsauth,
+            use_ssl=(parsed_url.scheme == "https"),
+            verify_certs=True,
+            connection_class=RequestsHttpConnection,
+        )
+
+        temp_connector = Connector(temp_os_client)
+        response = temp_connector.create_standalone_connector(payload)
+
+        print(response)
+        connector_id = response.get("connector_id")
+        return connector_id
+
+    def get_task(self, task_id, create_connector_role_name):
```

Comment:
Can't we reuse [this method](https://github.com/opensearch-project/opensearch-py-ml/blob/b26ea1ea360ac0169a0529859b41eee57ac80bf1/opensearch_py_ml/ml_commons/ml_commons_client.py#L389)?